docs.md

```diff
@@ -0,0 +1,98 @@
+- UserName
+- UserPrincipalName
+- GroupName
+- GroupPrincipalName
+
+In addition, as in the create validation, both a user subject and a group subject cannot be specified.
+## GlobalRole 
+
+### Validation Checks
+Note: all checks are bypassed if the GlobalRole is being deleted
+
+#### Escalation Prevention
+Users can only change GlobalRoles which have less permissions than they do. This is to prevents privilege escalation. 
+
+## GlobalRoleBinding 
+
+### Validation Checks
+Note: all checks are bypassed if the GlobalRoleBinding is being deleted
+
+#### Escalation Prevention
+Users can only create/update GlobalRoleBindings which grant permissions to GlobalRoles which have less permissions than they do. This is to prevents privilege escalation. 
+
+#### Valid Global Role Reference
+GlobalRoleBindings must refer to a valid global role (i.e. an existing `GlobalRole` object in the `management.cattle.io/v3` apiGroup).
+## ProjectRoleTemplateBinding 
+
+### Validation Checks
+#### Escalation Prevention
+Users can only create/update ProjectRoleTemplateBindings which grant permissions to RoleTemplates which have less permissions than they do. This is to prevents privilege escalation. 
+
+#### Invalid Fields - Create
+Users cannot create ProjectRoleTemplateBindings which violate the following constraints:
+- Either a user subject (through "UserName" or "UserPrincipalName") or a group subject (through "GroupName" or "GroupPrincipalName"), but not both, must be specified
+- A "ProjectName" must be specified
+- The roleTemplate indicated in "RoleTemplateName" must be:
+  - Valid (i.e. is an existing `roleTemplate` object in the `management.cattle.io/v3` apiGroup)
+  - Not locked (i.e. `roleTemplate.Locked` must be `false`)
+
+#### Invalid Fields - Update
+Users canot update the following fields after creation:
+- RoleTemplateName
+- ProjectName
+
+Users can update the following fields if they have not been set, but after they have been set they can't be changed:
+- UserName
+- UserPrincipalName
+- GroupName
+- GroupPrincipalName
+
+In addition, as in the create validation, both a user subject and a group subject cannot be specified.
+## RoleTemplate 
+
+### Validation Checks
+Note: all checks are bypassed if the RoleTemplate is being deleted
+
+####  Circular Reference
+Circular references to webhooks (a inherits b, b inherits a) are not allowed. More specifically, if "roleTemplate1" is included in the `roleTemplateNames` of "roleTemplate2", then "roleTemplate2" must not be included in the `roleTemplateNames` of "roleTemplate1". This checks prevents the creation of roles whose end-state cannot be resolved.
+
+#### Rules Without Verbs 
+Rules without verbs are not peritted. The `rules` included in a roleTemplate are of the same type as the rules used by standard kubernetes RBAC types (such as `Roles` from `rbac.authorization.k8s.io/v1`). Because of this, they inherit the same restrictions as these types, including this one.
```

Comment:
Thanks for catching the spelling error, will fix that. I will keep the suffix to this - I want to be clear that this restriction is because of the underlying restriction and I don't think that this is clear without that.

Reply:
Yeah, the suffix was just a nit. It appears redundant, but it doesn't hurt.